### ISSUE TITLE
3 x Failed Loop Outs - Temporary Failure

### ISSUE THREAD

**Issue by benniblitz** [lightninglabs/loop#513]
Raspiblitz v1.7.2
bitcoin v22.0.0
LND 0.14.2-beta

I need some help resolving 3 failed loop outs.

They all say Temporary Failure "The swap cannot continue due to an internal error. Manual intervention such as a restart is required." 

I have restarted my node and reinstalled LIT but the error persists and the HTLCs are still in pending. 

It seems the payments may have been split in two.

The 3 HTLCs and Hash Locks are: 
1. BenniBlitz - c-otto.de 5057244 sats e29fce7b83040f64f1b09746fd2d5a4c16518a2953dee5fc274c1d42970dc175
2. BenniBlitz - Kollider 5054482 sats 5c9b14f7bd7d88f89ef87d45404d8b18772f424cb2f2ee583febb28aeda34abe 
3. BenniBlitz - zero fee routing | CLN 3,959,599 sats a23202630d89802fffe9ad6824e5700be6caa5511cc8c4f9d571d502e0e183bc 

Any assistance is much appreciated!

![Initiated](https://user-images.githubusercontent.com/101555945/179695610-f5ca2ca7-b6a1-4171-a516-bcfe751b7cf0.png)

![Failure](https://user-images.githubusercontent.com/101555945/179695626-4c10f00d-fc56-4180-8230-594290226387.png)

**Comment by bhandras** [comment]
Hi, @benniblitz thanks for reporting. Could you please share the swap hash?

**Comment by benniblitz** [comment]
Hi @bhandras thanks for the reply! Sure thing, although could you please advise where and how can I access that information? I don't seem to be able to get it from the lightning terminal but perhaps I can access it via the command line. If so could you please share the exact command line prompts? Many thanks again

**Comment by benniblitz** [comment]
</head>

<body link="#0563C1" vlink="#954F72">



Swap ID | Type | Amount | Status | Created On | Updated On
-- | -- | -- | -- | -- | --
a23202630d89802fffe9ad6824e5700be6caa5511cc8c4f9d571d502e0e183bc | Loop Out | 3980000 | Failed | Jul 17, 4:36 PM | Jul 17, 4:36 PM
e29fce7b83040f64f1b09746fd2d5a4c16518a2953dee5fc274c1d42970dc175 | Loop Out | 5070000 | Failed | Jul 16, 9:14 PM | Jul 16, 9:14 PM
5c9b14f7bd7d88f89ef87d45404d8b18772f424cb2f2ee583febb28aeda34abe | Loop Out | 5070000 | Failed | Jul 16, 9:13 PM | Jul 16, 9:13 PM



</body>

</html>

**Comment by benniblitz** [comment]
Hi @bhandras thanks for the reply! Sure thing, if the information above is not sufficient could you please advise where and how can I access the additional information? I don't seem to be able to get it from the lightning terminal but perhaps I can access it via the command line. If so could you please share the exact command line prompts? Many thanks again

**Comment by bhandras** [comment]
@benniblitz from the server side logs I can see that in each case your invoice payments were accepted, the onchain htlcs were published, in the last case it's still there on-chain. Seems like your client doesn't finish the loop process for some reason. Can you please share your logs in a private way? They're in `~/.lit/logs/mainnet`. You can chat me on `keybase` if you'd like under the same handle to avoid leaking more PII.

**Comment by benniblitz** [comment]
2022-07-17 09:41:23.158 [ERR] LOOP: 5c9b14 Swap error: signing: rpc error: code = Unknown desc = selected sign method witness_v0 is not compatible with given pk script
2022-07-17 09:41:23.158 [INF] LOOP: 5c9b14 Loop out swap state: FailTemporary
2022-07-17 09:41:23.158 [ERR] LOOP: Execute error: signing: rpc error: code = Unknown desc = selected sign method witness_v0 is not compatible with given pk script

**Comment by benniblitz** [comment]
2022-07-17 09:41:23.278 [ERR] LOOP: e29fce Swap error: signing: rpc error: code = Unknown desc = selected sign method witness_v0 is not compatible with given pk script
2022-07-17 09:41:23.278 [INF] LOOP: e29fce Loop out swap state: FailTemporary
2022-07-17 09:41:23.279 [ERR] LOOP: Execute error: signing: rpc error: code = Unknown desc = selected sign method witness_v0 is not compatible with given pk script

**Comment by benniblitz** [comment]
> @benniblitz from the server side logs I can see that in each case your invoice payments were accepted, the onchain htlcs were published, in the last case it's still there on-chain. Seems like your client doesn't finish the loop process for some reason. Can you please share your logs in a private way? They're in `~/.lit/logs/mainnet`. You can chat me on `keybase` if you'd like under the same handle to avoid leaking more PII.

Thanks Andras, I just got in touch with you on keybase and sent the logs in a .txt file

**Comment by benniblitz** [comment]
The error has now changed to "On-chain Timeout: The on-chain HTLC did not confirm before its expiry, or it confirmed too late for us to reveal our preimage and claim."

**Comment by bhandras** [comment]
We figured out why your loop outs failed. You were using an old version of LiT (v0.6.4-alpha) where the recommended LND version is v0.14.2-beta. With LND v0.15.0-beta we changed how `SignOutputRaw` works where it didn't require the pkScript before it does now. In your case you updated your node to LND 0.15.0-beta while your loop client was still an old version, v0.15.1-beta (packaged with LiT). This was also fixed on the loop side (https://github.com/lightninglabs/loop/pull/469) and was released with loop version v0.18.0-beta which is compatible with LND 0.15.0-beta.

Once you update to the latest version of LiT, your loops should work. Unfortunately in the 3 cases you reported all the sweeps timed out on-chain, and we needed to sweep back those htlcs which justifies the 3x30k sats prepays.
